app/models/school.rb

```diff
@@ -28,6 +28,8 @@ class School < ApplicationRecord
     all_through: 7
   }.freeze
 
+  PHASE_STRINGS = PHASES.keys.map(&:to_s).index_with(&:itself).freeze
```

Comment:
Not a blocker but would probably rename this to `PHASES` (or inline it directly in the enum definition) and move the existing `PHASES` and `self.phase_code_to_enum` to `app/models/school_data_importer.rb` as that seems to be the only place it's used.
Same with the other X_STRINGS constants